controllers/storagecluster/cephobjectstores.go

```diff
@@ -42,10 +43,10 @@ func (obj *ocsCephObjectStores) ensureCreated(r *StorageClusterReconciler, insta
 	}
 	var cephObjectStores []*cephv1.CephObjectStore
 	// Add KMS details to cephObjectStores spec, only if
-	// cluster-wide encryption is enabled
-	// ie, sc.Spec.Encryption.ClusterWide/sc.Spec.Encryption.Enable is True
+	// cluster-wide encryption is enabled or any of the device set is encrypted
+	// ie, sc.Spec.Encryption.ClusterWide/sc.Spec.Encryption.Enable is True or any deviceSet is encrypted
 	// and KMS ConfigMap is available
-	if instance.Spec.Encryption.Enable || instance.Spec.Encryption.ClusterWide {
+	if instance.Spec.Encryption.Enable || instance.Spec.Encryption.ClusterWide || util.IsAnyDeviceSetEncrypted(instance.Spec.StorageDeviceSets) {
```

Comment:
We dont have objectstore at day-2?
But here we can also do a fallback mechanism

Reply:
As discussed this and other places where just KMS information is being added we don't need to respect the deviecSet setting first.